### ISSUE TITLE
Improve datapoint sync layout

### ISSUE THREAD

**Issue by valllllll2000** [akvo/akvo-flow-mobile#583]
We have decided to improve the usage of notifications as sometimes their use within the app is not correct and does not follow [google design guidelines](https://material.io/guidelines/patterns/notifications.html) at all.
Here is the original issue where all the notifications have been discussed
https://github.com/akvo/akvo-product-design/issues/218

For datapoint sync we have decided on the following specs:

- No notification will be used

- While syncing, there will be a loading spinning wheel (ProgressBar) displayed on top of the datapoint list or map.
- Success or failure will be displayed with a snackbar with some of the error messages will include a retry button. See https://docs.google.com/spreadsheets/d/1ONmVlA9TYHCDEkglfWqIbJse64DNmOyhrl7fFwdEReU/edit for more details and exact text specs.

**Comment by valllllll2000** [comment]
@janagombitova I would need your input on this.
![snackbar](https://cloud.githubusercontent.com/assets/923280/22432615/c5edd578-e715-11e6-8d23-fd7b6f577102.png)
As you can see on the picture, the message is too long for this small device. This happens for both messages:
* Sync failed. Flow dashboard assignment required
* Sync failed. Check your connection and try again

Here are my suggestions:
* Dashboard assignment missing
* Check your connection and retry

**Comment by janagombitova** [comment]
I personally like that the message says something went wrong. So here would be my suggestions: 
- Sync failed. Assignment missing. 
- Sync failed. No connection.

But if we agree that by giving the users the "retry" action, the notion that something went wrong is clear enough then fine. Then I would go for: 
- Dashboard assignment missing
- No connection 

@Kiarii is not available for Flow work so I brought him in the loop on the overall notifications improvement project we started. Thus, Kiarii what are your thoughts? These are the first changes for the orange coloured current notifications.

**Comment by Kiarii** [comment]
If Sync refers to the transfer of data from phone to server, i think it ought to take a simpler approach as you mentioned (with the ticking). 

I also think a Synch failure may occur if the server is down; a situation in which the user cannot intervene from the app. Thus it may be best that the app handles 'Retry' (until success) and not to prevent the user from doing other things or even asking them to initiate the task that the system can/shd handle.

This is my first take, trying to get a good feel on the notifications..

**Comment by valllllll2000** [comment]
@janagombitova I will then be using the strings, they fit in one line perfectly.
Sync failed. Assignment missing.
Sync failed. No connection.

@Kiarii this task is only about getting the datapoints in the device so getting data from server to device. In this case showing a snackbar as shown in the image does not block the ui, it disappears after a few seconds and also the user can navigate away from that window without anything blocking him.

**Comment by Kiarii** [comment]
![sync data](https://cloud.githubusercontent.com/assets/12692444/22831002/5c060be0-efb1-11e6-8e70-cad198a1beeb.png)

* This is what we have agreed to for now..

**Comment by janagombitova** [comment]
https://cloud.githubusercontent.com/assets/12692444/22882604/adfecc80-f1f4-11e6-8c16-ad66a31babc8.png

Here is the mockup of the behaviour for monitored surveys that @Kiarii shared in this issue https://github.com/akvo/akvo-flow-mobile/issues/621 but is relevant for the work here.

**Comment by valllllll2000** [comment]
## Test plan
1) Install fresh version of the app (no update)
2) Select one of your assigned surveys and press sync while in list tab
You should see a "loading". 
If there are no datapoints it will just stop syncing and not show any errors just the usual empty screen
If there are datapoints, they will start coming in. At the end, you will get a snackbar with the number of datapoints synced.
If possible try with a survey which has more than 300 datapoints. 
3) If you want to see something meaning full in the map tab, you need surveys with location (use as datapoint location is checked)

**Comment by valllllll2000** [comment]
These are the new layouts for the datapoints list/map:
* empty screens are covered in issue #621 
* When pressed "sync" we have a blue progressBar shown either over the empty screen or already existing datapoints:
![screenshot_20170705-123131](https://user-images.githubusercontent.com/923280/27860994-36fcf512-617f-11e7-8ad2-7c543a76aecd.png)

* All datapoints successfully synced, we display a snackbar with the total number of items synced:
![screenshot_20170705-123137](https://user-images.githubusercontent.com/923280/27861033-64461058-617f-11e7-88b5-7d55f40f82f3.png)

* In case of error, the error message is displayed in the snackbar with, if relevant, a retry button:
![screenshot_20170705-123159](https://user-images.githubusercontent.com/923280/27861067-84333fc6-617f-11e7-97c9-47efb1425c8b.png)
![screenshot_1499251429](https://user-images.githubusercontent.com/923280/27861092-a0670664-617f-11e7-9407-f2bcb055c84b.png)

**Comment by valllllll2000** [comment]
Here is the test plan again to avoid searching for it in the comments.

## Test plan
1) Install fresh version of the app (no update)
2) Select one of your assigned surveys and press sync while in list tab
You should see a "loading". 
If there are no datapoints it will just stop syncing and not show any errors just the usual empty screen
If there are datapoints, they will start coming in. At the end, you will get a snackbar with the number of datapoints synced.
If possible try with a survey which has more than 300 datapoints. 
3) If you want to see something meaningfull in the map tab, you need surveys with location (use as datapoint location is checked)
4) Possible errors: 
    - No internet (flight mode)
    - Sync disabled in settings + no wifi only 3g
    - Assignment missing
    - Other errors (generic or server errors)

**Comment by janagombitova** [comment]
As we spoke about yesterday, I cannot sync in data points on the 2.4.0 version. Will continue testing once you say we can give it another try.

**Comment by janagombitova** [comment]
Hey @valllllll2000 I tried syncing with the latests apk on uat1 - points do sync. But when I tried a survey with 1017 datapoints only 300 arrived to my phone and then the syncing stopped. With the 300 points on my phone I synced again after I received again the snack bar that 300 points synced in, after checking the phone stats I had 599 points on my phone. 

So with each new user triggered sync a new batch of points came in, but not all points got synced at once what is the expectation.
